backend/src/main/java/ai/verta/modeldb/monitoring/AuditLogInterceptor.java

```diff
@@ -0,0 +1,72 @@
+package ai.verta.modeldb.monitoring;
+
+import io.grpc.Context;
+import io.grpc.Contexts;
+import io.grpc.ForwardingServerCall;
+import io.grpc.ForwardingServerCallListener;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import io.prometheus.client.Counter;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class AuditLogInterceptor implements ServerInterceptor {
+
+  private static final Logger LOGGER = LogManager.getLogger(AuditLogInterceptor.class);
+  public static final Context.Key<AtomicInteger> CALL_COUNT = Context.key("audit_count");
+  public static final AtomicInteger auditLogCount = new AtomicInteger(0);
```

Comment:
this should have instance for every thread (to be created in interceptCall method)

Reply:
Yep